### PR TITLE
Introduce ay api-client and composables

### PR DESCRIPTION
**Pull request by I2olanD**
- [x] I Accept [Individual Contributor License Agreement](https://github.com/DivanteLtd/next/blob/master/CLA.md)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -12,6 +12,8 @@
     "build:core:factories": "cd packages/core/factories && yarn build",
     "build:core": "yarn build:core:interfaces && yarn build:core:utils && yarn build:core:factories",
     "build:prismic": "cd packages/prismic && yarn build",
+    "build:ay:api-client": "cd packages/about-you/api-client && yarn build",
+    "build:ay:composables": "cd packages/about-you/composables && yarn build",
```
Comment (lukeromanowicz) [inline package.json:16]: Could we also add `build:ay` ?

File: packages/about-you/api-client/tests/api/getCategory/index.spec.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import getCategory from '../../../src/api/getCategory';
+import { apiClient } from '../../../src/index';
+
+const mockGetByIds = jest.fn((params) => params);
+const mockGetByPath = jest.fn((params) => params);
+const mockGetRoots = jest.fn((params) => params);
+
+apiClient.categories.getByIds = mockGetByIds;
+apiClient.categories.getByPath = mockGetByPath;
```
Comment (lukeromanowicz) [inline packages/about-you/api-client/tests/api/getCategory/index.spec.ts]: This isn't the right way to mock apiClient. Please use jest.mock for that.

File: packages/about-you/api-client/tests/api/getProduct/index.spec.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+const mockGetById = jest.fn((params) => params);
+const mockGetByIds = jest.fn((params) => params);
+const mockQuery = jest.fn((params) => params);
+
+const mockGetByKey = jest.fn((params) => params);
+const mockSuggestions = jest.fn((params) => params);
+
+apiClient.products.getById = mockGetById;
+apiClient.products.getByIds = mockGetByIds;
+apiClient.products.query = mockQuery;
```
Comment (lukeromanowicz) [inline packages/about-you/api-client/tests/api/getProduct/index.spec.ts]: Same here

File: packages/about-you/api-client/tests/setup.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import { setup } from './../src/index';
+
+jest.mock('@aboutyou/backbone', () => ({
+  BapiClient: jest.fn(() => ({
```
Comment (lukeromanowicz) [inline packages/about-you/api-client/tests/setup.ts]: I don't think you need it to be executed in every test...

**Review by filrak — DISMISSED** [review]
// woops

**Review by filrak — APPROVED** [review]

**Comment by filrak** [comment]
Once @lukeromanowicz desires are fulfilled I will merge that. Thanks @I2olanD ! Really solid job on that .

Just please add AY tests check to travis

**Pull request closed without merging**